programs/mmm/src/instructions/mpl_core_asset/mpl_core_fulfill_buy.rs

```diff
@@ -0,0 +1,379 @@
+
+        &remaining_accounts[2..]
+    } else {
+        remaining_accounts
+    };
+
+    if pool.reinvest_fulfill_buy {
+        if pool.using_shared_escrow() {
+            return Err(MMMErrorCode::InvalidAccountState.into());
+        }
+
+        let transfer_asset_builder = TransferV1Builder::new()
+            .asset(asset.key())
+            .payer(payer.key())
+            .collection(
+                if let UpdateAuthority::Collection(collection_address) = asset.update_authority {
+                    Some(collection_address)
+                } else {
+                    None
+                },
+            )
+            .new_owner(pool.key())
+            .instruction();
+
+        let mut account_infos = vec![
+            asset.to_account_info(),
+            pool.to_account_info(),
+            payer.to_account_info(),
+        ];
+        if collection.key != &Pubkey::default() {
+            if UpdateAuthority::Collection(collection.key()) != asset.update_authority {
+                return Err(MMMErrorCode::InvalidAssetCollection.into());
+            }
+            account_infos.push(collection.to_account_info());
+        }
+
+        invoke(&transfer_asset_builder, account_infos.as_slice())?;
+
+        pool.sellside_asset_amount = pool
+            .sellside_asset_amount
+            .checked_add(1)
+            .ok_or(MMMErrorCode::NumericOverflow)?;
+        sell_state.pool = pool.key();
+        sell_state.pool_owner = owner.key();
+        sell_state.asset_mint = asset.key();
+        sell_state.cosigner_annotation = pool.cosigner_annotation;
+        sell_state.asset_amount = sell_state
+            .asset_amount
+            .checked_add(1)
+            .ok_or(MMMErrorCode::NumericOverflow)?;
+    } else {
+        let transfer_asset_builder = TransferV1Builder::new()
+            .asset(asset.key())
+            .payer(payer.key())
+            .collection(
+                if let UpdateAuthority::Collection(collection_address) = asset.update_authority {
+                    Some(collection_address)
+                } else {
+                    None
+                },
+            )
+            .new_owner(owner.key())
+            .instruction();
+
+        let mut account_infos = vec![
+            asset.to_account_info(),
+            owner.to_account_info(),
+            payer.to_account_info(),
+        ];
+        if collection.key != &Pubkey::default() {
+            if UpdateAuthority::Collection(collection.key()) != asset.update_authority {
+                return Err(MMMErrorCode::InvalidAssetCollection.into());
+            }
+            account_infos.push(collection.to_account_info());
+        }
+
+        invoke(&transfer_asset_builder, account_infos.as_slice())?;
+    }
+
+    // pool owner as buyer is going to pay the royalties
+    let royalty_paid = if let Some(metadata) = &metadata {
+        pay_creator_fees_in_sol(
+            pool.buyside_creator_royalty_bp,
```

Comment:
since we are enforcing royalties, we should set this to constant 10_000 like what we do for pNfts